vars/publishDistributionBuildResults.groovy

```diff
@@ -0,0 +1,170 @@
+                                version, distributionBuildNumber, distributionBuildUrl,
+                                buildStartTime, rc, rcNumber, componentCategory, "passed"
+                                )
+            finalJsonDoc += "{\"index\": {\"_index\": \"${indexName}\"}}\n${jsonData}\n"
+        }
+    }
+    writeFile file: "test-records.json", text: finalJsonDoc
+    def fileContents = readFile(file: "test-records.json").trim()
+    indexFailedTestData(indexName, "test-records.json")
+
+}
+
+void indexFailedTestData(indexName, testRecordsFile) {
+    withCredentials([
+            string(credentialsId: 'jenkins-health-metrics-account-number', variable: 'METRICS_HOST_ACCOUNT'),
+            string(credentialsId: 'jenkins-health-metrics-cluster-endpoint', variable: 'METRICS_HOST_URL')
+    ]) {
+        withAWS(role: 'OpenSearchJenkinsAccessRole', roleAccount: "${METRICS_HOST_ACCOUNT}", duration: 900, roleSessionName: 'jenkins-session') {
+            def awsAccessKey = env.AWS_ACCESS_KEY_ID
+            def awsSecretKey = env.AWS_SECRET_ACCESS_KEY
+            def awsSessionToken = env.AWS_SESSION_TOKEN
+            sh """
+                set +e
+                set +x
+                echo "INDEX NAME IS ${indexName}"
+                INDEX_MAPPING='{
+                    "mappings": {
+                        "properties": {
+                            "component": {
+                                "type": "keyword"
+                            },
+                            "component_repo": {
+                                "type": "keyword"
+                            },
+                            "component_ref": {
+                                "type": "keyword"
+                            },
+                            "version": {
+                                "type": "keyword"
+                            },
+                            "distribution_build_number": {
+                                "type": "integer"
+                            },
+                            "distribution_build_url": {
+                                "type": "keyword"
+                            },
+                            "build_start_time": {
+                                "type": "date",
+                                "format": "epoch_millis"
+                            },
+                            "rc": {
+                                "type": "keyword"
+                            },
+                            "rc_number": {
+                                "type": "integer"
+                            },
+                            "component_category": {
+                                "type": "keyword"
+                            },
+                            "component_build_result": {
+                                "type": "keyword"
+                            }
+                        }
+                    }
+                }'
```

Comment:
Should we define this JSON in groovy using JSON parses? I believe shell wont be able to catch the typos

Reply:
Initially I have tried that Sayali, but end of the day its the index mapping using during `curl` with `-d`, with groovy the curl is rejecting the data parsing with format exception, hence moved the block to shell in a proper json format and then adding with via `-d` had no parsing issues. Added some good tests with normalized data `normalizeString(calledCommands[0]) == normalizeString(expectedCommandBlock)`, if the index mapping is changed lets says with typos the tests will fail and and as well OpenSearch itself will fail to index once the index mapping is created.